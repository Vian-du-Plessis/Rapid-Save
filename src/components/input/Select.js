import React from 'react';
import styles from './Select.module.scss'

const Select = ( props ) => {
    return (
        <select>
            <option>{ props.defaultValue || '10%' }</option>
        </select>
    );
};

export default Select;