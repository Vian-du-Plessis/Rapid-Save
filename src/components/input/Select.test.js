import Select from './Select';
import TestRenderer from 'react-test-renderer';
import { act, create } from "react-test-renderer";


describe("Test the Select component ", () =>{
    test("Create Rendered Select", () => {

        const testBtn = TestRenderer.create(
            <Select
                defaultValue='10%'
            />
        );
        const btnTestInstance = testBtn.root;
        expect(btnTestInstance.findByType(Select).props.defaultValue).toBe["10%"];
    });

    test("Select Matches Snapshot", async () => {
        let root;
        await act(() => {
            root = create(            
                <Select
                    defaultValue='10%'
                />
            );
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(
                <Select
                    defaultValue='10%'
                />
            );
            expect(root.toJSON()).toMatchSnapshot();
        })
    })
})
