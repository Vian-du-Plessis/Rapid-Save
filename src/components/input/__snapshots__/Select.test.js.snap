// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Select component  Select Matches Snapshot 1`] = `
<select>
  <option>
    10%
  </option>
</select>
`;

exports[`Test the Select component  Select Matches Snapshot 2`] = `
<select>
  <option>
    10%
  </option>
</select>
`;
